biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java

```diff
@@ -1210,4 +1210,14 @@ public void removePrefix(String prefixLow) {
 		directories.subMap(prefixLow, prefixHigh)
 			.clear();
 	}
+
+	public void removeSubDirs(String prefixLow) {
+		String prefixHigh = prefixLow + "\uFFFF";
+		if (prefixLow.endsWith("/")) {
+			prefixLow = prefixLow.substring(0, prefixLow.length() - 1);
+			prefixHigh = prefixLow + "\uFFFF";
+		}
+		directories.subMap(prefixLow, prefixHigh)
```

Comment:
This seems wrong. You always must modify both resources and directories. It is wrong to only clear things out of directories and not resources. You also need to clear the resources in those directories from resources.

Reply:
yeah, you're right. My previous logic was better and did `removePrefx()` on each subdir dir. I'll re-revert to that. thx

Reply:
Updated